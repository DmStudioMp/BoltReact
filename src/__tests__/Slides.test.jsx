import React from "react";
import { render, screen } from "@testing-library/react";
import Slide, { SlideImg } from "../Components/Carrousel/Slides/Slides.jsx";
import "@testing-library/jest-dom";

describe("Slide Component", () => {
  test("renders Slide component without crashing", () => {
    render(<Slide />);
    expect(screen.queryByRole("presentation")).not.toBeNull();
  });

  test("applies 'active' class when active prop is true", () => {
    render(
      <Slide active={true}>
        <p>Active Slide</p>
      </Slide>
    );
    const slideElement = screen
      .getByText("Active Slide")
      .closest(".carousel-item");
    expect(slideElement).toHaveClass("active");
  });

  test("does not apply 'active' class when active prop is false", () => {
    render(
      <Slide active={false}>
        <p>Inactive Slide</p>
      </Slide>
    );
    const slideElement = screen
      .getByText("Inactive Slide")
      .closest(".carousel-item");
    expect(slideElement).not.toHaveClass("active");
  });

  test("renders children correctly", () => {
    render(
      <Slide>
        <p>Slide Content</p>
      </Slide>
    );
    expect(screen.getByText("Slide Content")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <Slide active={true}>
        <p>Snapshot Slide</p>
      </Slide>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("SlideImg Component", () => {
  test("renders SlideImg component without crashing", () => {
    render(<SlideImg />);
    const svgElement = screen.getByRole("img", { hidden: true });
    expect(svgElement).toBeInTheDocument();
    expect(svgElement).toHaveAttribute("xmlns", "http://www.w3.org/2000/svg");
  });

  test("renders placeholder rectangle with correct dimensions and fill", () => {
    render(<SlideImg />);
    const rectElement = screen
      .getByRole("img", { hidden: true })
      .querySelector("rect");
    expect(rectElement).toHaveAttribute("width", "100%");
    expect(rectElement).toHaveAttribute("height", "100%");
    expect(rectElement).toHaveAttribute("fill", "#777");
  });

  test("matches snapshot for SlideImg", () => {
    const { container } = render(<SlideImg />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
