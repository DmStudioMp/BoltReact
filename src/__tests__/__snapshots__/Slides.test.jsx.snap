// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicators Component matches the snapshot for visual regression testing 1`] = `
<div>
  <div
    class="carousel-indicators justify-content-center"
    role="tablist"
  >
    <button
      aria-label="Go to slide 1"
      aria-selected="true"
      class="indicator-btn"
      type="button"
    >
      1
    </button>
    <button
      aria-label="Go to slide 2"
      aria-selected="false"
      class="indicator-btn"
      type="button"
    >
      2
    </button>
    <button
      aria-label="Go to slide 3"
      aria-selected="false"
      class="indicator-btn"
      type="button"
    >
      3
    </button>
  </div>
</div>
`;

exports[`Slide Component matches snapshot 1`] = `
<div
  class="carousel-item active"
  role="tabpanel"
>
  <div
    class="d-block w-100"
  >
    <p>
      Snapshot Slide
    </p>
  </div>
</div>
`;

exports[`SlideImg Component matches snapshot for SlideImg 1`] = `
<svg
  aria-hidden="true"
  class="bd-placeholder-img"
  focusable="false"
  height="100%"
  preserveAspectRatio="xMidYMid slice"
  role="img"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="#777"
    height="100%"
    width="100%"
  />
</svg>
`;
